### PR TITLE
ht 3

### PR DESCRIPTION
**Pull request by manhengest**

### CODE REVIEW

File: src/components/Buttons/interface.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export interface ButtonProps {
+    cb(): void,
```
Comment (SteveRusin) [inline src/components/Buttons/interface.ts:2]: use more specific prop name for `cb`
`cb` -> `clickHander`

**Review by SteveRusin — APPROVED** [review]

File: src/components/Buttons/PrimaryButton/index.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import { ButtonProps } from "../interface";
+import "./primary-button.scss"
+
+export const PrimaryButton:React.FunctionComponent<ButtonProps> = ({ clickHandler, title }) => {
+    return (
+        <button className="primary-button" onClick={ () => clickHandler() }>
```
Comment (ygorbunkov) [inline src/components/Buttons/PrimaryButton/index.tsx:7]: No need to wrap into extra (anonymous) arrow function :
```jsx
<button className="primary-button" onClick={clickHandler} >
```

File: src/components/MovieCard/models.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export interface MovieCardProps {
+    posterUrl: string,
+    link: string,
+    title: string,
+    genre: string,
```
Comment (ygorbunkov) [inline src/components/MovieCard/models.ts:5]: Range of values for `genre` property is quite limited, so, for type safety, you may constrain those, using `enum`:
```ts
export enum Genre {
   ACTION = 'action',
   COMEDY = 'comedy'
   ..
}

export interface Movie {
   ..
   genre: Genre,
   ..
}
```

File: src/components/PrimaryTitle/index.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+interface PrimaryTitleProps {
+    title: string,
+    additionalClass?: string
+}
+
+export const PrimaryTitle:React.FunctionComponent<PrimaryTitleProps> = ({ title, additionalClass }) => {
+    const classes = ["primary-title"]
+
+    if (additionalClass) {
```
Comment (ygorbunkov) [inline src/components/PrimaryTitle/index.tsx:12]: You may consider alternative technique to manipulate `className` conditionally (e.g. based on props):
```jsx
..
const titleClassName = [
   'primary-title',
   ...(additionalClass ? [`primary-title_${additionalClass}`] : [])
].join(' ')

return (
   <span className={titleClassName}>{title}</span>
)
```

File: src/containers/Genres/index.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import "./genres.scss"
+import { GenreItem } from "../../components/GenreItem";
+
+export const Genres:React.FunctionComponent = () => {
+    const genres = [
```
Comment (ygorbunkov) [inline src/containers/Genres/index.tsx:6]: Just another way to do the same:
```js
['All', 'Documentary', 'Comedy', 'Horror', 'Crime']
   .map((title, _id) => ({
      title,
      id: _id+1,
      active: !_id
   }))
```

Or, better, yet, if you would strong-type genres (using `enum Genre`), something like following would work:
```ts
Object
   .values(Genre)
   .map((title, _id) => ({
      title,
      id: _id+1,
      active: !_id
   }))
```

**Pull request merged**